core/src/wallet/algorand/AlgorandBlockchainObserver.cpp

```diff
@@ -0,0 +1,158 @@
+                if (account->putTransaction(sql, tx) != Account::FLAG_TRANSACTION_IGNORED)
+                    account->emitEventsNow();
+            });
+        }
+    }
+
+    void BlockchainObserver::putBlock(const api::Block &block) {
+        std::lock_guard<std::mutex> lock(_lock);
+        for (const auto &account : _accounts) {
+            account->run([account, block]() {
+                bool shouldEmitNow = false;
+                {
+                    soci::session sql(account->getWallet()->getDatabase()->getPool());
+                    shouldEmitNow = account->putBlock(sql, block);
+                }
+                if (shouldEmitNow)
+                    account->emitEventsNow();
+            });
+        }
+    }
+
+
+    void BlockchainObserver::onStart() {
+        connect();
+    }
+
+    void BlockchainObserver::onStop() {
+        auto self = shared_from_this();
+        run([self]() {
+            if (self->_socket != nullptr)
+                self->_socket->close();
+            self->_handler = [self](WebSocketEventType event,
+                                    const std::shared_ptr<WebSocketConnection> &connection,
+                                    const Option<std::string> &message, Option<api::ErrorCode> code) {
+
+            };
+        });
+    }
+
+    void BlockchainObserver::connect() {
+        AlgorandAbstractBlockchainObserver::logger()->info("Connect {} observer", getCurrency().name);
+        auto self = shared_from_this();
+        _handler = [self](WebSocketEventType event,
+                            const std::shared_ptr<WebSocketConnection> &connection,
+                            const Option<std::string> &message, Option<api::ErrorCode> code) {
+            self->onSocketEvent(event, connection, message, code);
+        };
+        _client->connect(_url, _handler);
+    }
+
+    void BlockchainObserver::reconnect() {
+        auto self = shared_from_this();
+        auto delay = std::min(std::max(Fibonacci::compute(_attempt) * 500, 30000), 500);
+        AlgorandAbstractBlockchainObserver::logger()->info("Attempt reconnection in {}ms for {} observer", delay,
+                                                getCurrency().name);
+        getContext()->delay(LambdaRunnable::make([self]() {
+            self->connect();
+        }), delay);
+    }
+
+
+    void BlockchainObserver::onMessage(const std::string &message) {
+        auto self = shared_from_this();
+        run([message, self]() {
+            // FIXME Needed or not?
+            /*
+            auto result = JSONUtils::parse<AlgorandWebSocketNotificationParser>(message);
+            result->block.currencyName = self->getCurrency().name;
+            if (result->type == "transaction") {
+                self->putTransaction(result->transaction);
+            } else if (result->type == "block") {
+                self->putBlock(result->block);
+            }
+            */
```

Comment:
Oops sorry for leaving that comment, we forgot about it!

That's a question about the full class actually!
Commenting this small piece of code actually disables the whole functionality of the Observer, since it doesn't react to any event.

A bit of context: we implemented that observer "just in case", because all other coins appear to have one. However we were not sure if it's useful, and I think it's not since we still don't use it today.

I think we should remove it completely, unless you see a mandatory need of an Observer for any coin?...

Reply:
@haammar-ledger I would say unused code is the worst pain for a developer so I suggest to delete it. However @pollastri-pierre what do you think about it ?

Reply:
(Solved by removing AlgorandBlockchainObserver)